bundle/config/mutator/python/override.go

```diff
@@ -0,0 +1,313 @@
+		leftMapping := left.MustMap()
+		rightMapping, err := dyn.GetByPath(rightRoot, p)
+
+		if err != nil {
+			return dyn.InvalidValue, err
+		}
+
+		out, _, err := overrideMapping(
+			leftMapping,
+			rightMapping.MustMap(),
+			func(left dyn.Value, right dyn.Value) (dyn.Value, isModified, error) {
+				merged, modified, err := override(left, right)
+
+				if err != nil {
+					return dyn.InvalidValue, NotModified, err
+				}
+
+				if modified == Modified {
+					if mode == OverrideModeAppend && left != dyn.NilValue {
+						return dyn.InvalidValue, NotModified, unexpectedChangeError(left.Location())
+					}
+
+					if mode == OverrideModeAppendOrModify && right == dyn.NilValue {
+						return dyn.InvalidValue, NotModified, unexpectedChangeError(left.Location())
+					}
+				}
+
+				return merged, modified, nil
+			})
+
+		if err != nil {
+			return dyn.InvalidValue, err
+		}
+
+		return dyn.NewValue(out, left.Location()), nil
+	})
+}
+
+func overrideMapping(leftMapping dyn.Mapping, rightMapping dyn.Mapping, overrideFunc overrideFunc) (dyn.Mapping, isModified, error) {
+	out := dyn.NewMapping()
+	var modified = false
+
+	for _, leftPair := range leftMapping.Pairs() {
+		// detect if key was removed
+		if _, ok := rightMapping.GetPair(leftPair.Key); !ok {
+			out, valueModified, err := overrideFunc(leftPair.Value, dyn.NilValue)
+
+			if err != nil {
+				return dyn.NewMapping(), NotModified, err
+			}
+
+			if out != dyn.NilValue {
+				return dyn.NewMapping(), NotModified, fmt.Errorf("'overrideFunc' didn't return return Nil")
+			}
+
+			modified = modified || valueModified.value
+		}
+	}
+
+	// iterating only right mapping will remove keys not present anymore
+	// and insert new keys
+
+	for _, rightPair := range rightMapping.Pairs() {
+		var leftValue dyn.Value
+		var key dyn.Value
+
+		if leftPair, ok := leftMapping.GetPair(rightPair.Key); ok {
+			leftValue = leftPair.Value
+
+			// key was there before, so keep its location
+			key = leftPair.Key
+		} else {
+			leftValue = dyn.NilValue
+			key = rightPair.Key
+
+			// always modified because we did insert here
+			modified = true
+		}
+
+		newValue, keyModified, err := overrideFunc(leftValue, rightPair.Value)
+		modified = modified || keyModified.value
+
+		if err != nil {
+			return dyn.NewMapping(), NotModified, err
+		}
+
+		if newValue == dyn.NilValue {
+			continue
+		}
+
+		err = out.Set(key, newValue)
+
+		if err != nil {
+			return dyn.NewMapping(), NotModified, err
+		}
+	}
+
+	if modified {
+		return out, Modified, nil
+	} else {
+		return leftMapping, NotModified, nil
+	}
+}
+
+func override(left dyn.Value, right dyn.Value) (dyn.Value, isModified, error) {
+	if left == dyn.NilValue {
+		if right == dyn.NilValue {
+			return dyn.NilValue, NotModified, nil
+		} else {
+			return right, Modified, nil
+		}
+	}
+
+	if left.Kind() != right.Kind() {
+		return right, Modified, nil
+	}
+
+	if left.Kind() == dyn.KindMap {
```

Comment:
Could you turn this into a switch/case with a function call for each kind? Right now the implementation is split between a chunk here and a chunk in the function being called.